contracts/proposals/ERC1046/TokenMetadata.sol

```diff
@@ -0,0 +1,34 @@
+pragma solidity ^0.4.21;
+
+import "../../token/ERC20/ERC20.sol";
+
+
+/**
+ * @title ERC-1047 Token Metadata
+ * @dev See https://eips.ethereum.org/EIPS/eip-1046
+ * @dev tokenURI must respond with a URI that implements https://eips.ethereum.org/EIPS/eip-1047
+ * @dev TODO - update https://github.com/OpenZeppelin/openzeppelin-solidity/blob/master/contracts/token/ERC721/ERC721.sol#L17 when 1046 is finalized
+ */
+contract ERC20TokenMetadata is ERC20 {
+  function tokenURI() external view returns (string);
+}
+
+
+contract ERC20WithMetadata is ERC20TokenMetadata {
+  string private tokenURI_ = "";
+
+  function ERC20WithMetadata(string _tokenURI)
+    public
+  {
+    tokenURI_ = _tokenURI;
+  }
+
+  function tokenURI() external view returns (string) {
+    // in the effort of minimum-effort compatibility,
+    //   tokenURI MUST throw if not exists https://eips.ethereum.org/EIPS/eip-721
```

Comment:
I don't understand how this is about compatibility. ER721 "throws if `_tokenId` is not a valid NFT". There's no `_tokenId` argument here, but as I understand it there is an implicit token id which is the token address, which sounds like a "valid token" under any interpretation.

I guess an empty string can never be a valid URI though. If that is the reason though, IMO we should check this in the setter of `tokenURI_` and not when returning it. Or not check it at all, because we're not going to validate that it's a valid URI anyway.

Reply:
Yeah you're right, this check doesn't really make sense. Will update